woocommerce/woocommerce-gateway-payfast
# Implement Playwright

### Description


This pull request implements the Playwright E2E test framework, including tests for critical flows.

#### List of E2E Tests
- [x] [Plugin Activation Without Errors](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#plugin-activation-without-errors)
- [x] [Can see a message when the core plugin (WooCommerce) is not active](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#can-see-a-message-when-the-core-plugin-woocommerce-is-not-active)
- [x] [Payfast Gateway Disabled with Non-ZAR Currency](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#payfast-gateway-disabled-with-non-zar-currency)
- [x] [Payfast Plugin Compatibility with ZAR Store Currency](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#payfast-plugin-compatibility-with-zar-store-currency)
- [x] [Enable Payfast Option](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#enable-payfast-option)
- [x] [Title and Description Display](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#title-and-description-display)
- [x] [Verify Merchant ID, Merchant Key, and Passphrase](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-merchant-id-merchant-key-and-passphrase)
- [x] [Verify Debug Email Configuration](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-debug-email-configuration)
- [x] [Verify Enable Logging Functionality](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-enable-logging-functionality)
- [ ] <s>[Verify Logging when "Enable Logging" is Not Checked](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-logging-when-enable-logging-is-not-checked)</s>: Log creates for test transactions.
- [x] [Verify Payfast Payment Option Availability](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-payfast-payment-option-availability)
- [x] [Verify Payfast Option Visibility](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-payfast-option-visibility)
- [x] [Verify Payfast Payment Process](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-payfast-payment-process)
- [x] [Verify Payfast Payment Confirmation](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-payfast-payment-confirmation)
- [x] [Verify Payfast Payment Failure Handling](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-payfast-payment-failure-handling)
- [ ] <s> [Verify ITN for Successful Payment](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#verify-itn-for-successful-payment)</s> Manual Test. In E2E tests, we are simulating ITN.
- [x] [Test Payfast Payment Method on Block-Based Checkout](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#test-payfast-payment-method-on-block-based-checkout)
 - [x] [Test Subscription Product with PayFast Payment Gateway](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#test-subscription-product-with-payfast-payment-gateway)
- [x] [Renew Subscription via PayFast](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Testing-instructions-for-critical-flows#renew-subscription-via-payfast)

### Steps to Test



 - E2E GitHub workflow behaves as described in the issue description
 - E2E tests should pass locally. `npm run env:start-local && npm run test:e2e-local`

### Documentation





### Changelog Entry

> Dev - Add Playwright end-to-end tests.

Closes #128.

### Todo
- [ ] After pull request merge, update the E2E test status of each test in [the critical flow table](https://github.com/woocommerce/woocommerce-gateway-payfast/wiki/Critical-Flows).
